src/main/java/org/opentripplanner/middleware/models/MonitoredTrip.java

```diff
@@ -128,5 +201,49 @@ public boolean canBeManagedBy(Auth0UserProfile user) {
         return super.canBeManagedBy(user);
     }
 
+    private Bson tripIdFilter() {
+        return eq("monitoredTripId", this.id);
+    }
+
+    /**
+     * Get the journey state for this trip.
+     */
+    public JourneyState retrieveJourneyState() {
+        JourneyState journeyState = Persistence.journeyStates.getOneFiltered(tripIdFilter());
+        // If journey state does not exist, create and persist.
+        if (journeyState == null) {
```

Comment:
There needs to be some kind of check for when the MonitoredTrip is updated or deleted. If an update or delete occurs the associated JourneyState probably needs to deleted as well, or maybe updated if the new itinerary associated with the updated MonitoredTrip is close enough that we don't want to send off certain notifications that still apply.